_blogposts/compiler/overview-of-new_encoding.mdx

```diff
@@ -0,0 +1,229 @@
+---
+author: hongbo
+date: "2020-06-10 18:00" 
+previewImg:
+category: compiler
+title: Overview of new encoding in BuckleScript
+description: |
+  Highlights of our newest changes to the internal representation of lazy values
+  and how it will benefit our users.
+---
+
+
+---
+title: An overview of new data representation in BuckleScript
+---
+
+In the next version of BuckleScript, we made several major changes to tweak the data representation for various data types, it's more idiomatic and debugger friendly. 
+
```

Comment:
...we made several major changes to tweak the data representation for various data types, making them more idiomatic and debugger friendly.